lib/vagrant.pm

```diff
@@ -33,9 +43,8 @@ sub run_vagrant_cmd {
 
     my $logfile = 'vagrant_cmd.log';
     local $@;
-    my $ret = eval {
-        return script_run("VAGRANT_LOG=DEBUG vagrant $cmd 2> $logfile", %args);
-    };
+    my $ret = script_run("VAGRANT_LOG=DEBUG vagrant $cmd 2> $logfile", %args);
+
     return undef if $ret == 0;
     if (!$ret) {
```

Comment:
The bug is probably here

Reply:
the `if (!$ret)` was not working for reasons. It's working now.

Reply:
Because

```
    return undef if $ret == 0;
    if (!$ret) {
    // unreachable!
```